bundle/edu.gemini.pot/src/main/scala/edu/gemini/spModel/target/env/OptsList.scala

```diff
@@ -0,0 +1,152 @@
+    })
+
+  def contains(a: A): Boolean =
+    toList.contains(a)
+
+  def focus: Option[A] =
+    toDisjunction.toOption.map(_.focus)
+
+  def focusIndex: Option[Int] =
+    toDisjunction.toOption.map(_.lefts.length)
+
+  /** Moves or sets the focus to `a` if `a` is a member of the options list and
+    * returns the updated `OptsList` in a `Some`. Otherwise, returns `None`.
+    */
+  def focusOn(a: A)(implicit ev: Equal[A]): Option[OptsList[A]] =
+    toList.span(_ =/= a) match {
+      case (_, Nil)                 => none
+      case (lefts, focus :: rights) => some(focused(lefts, focus, rights))
+    }
+
+  /** Sets the focus element to the element at the given index if in range,
+    * returning a new `OptsList` wrapped in a `Some`.  Otherwise, returns `None`.
+    */
+  def focusOnIndex(i: Int): Option[OptsList[A]] =
+    if (i < 0) none
+    else toList.splitAt(i) match {
+      case (_, Nil)                 => none
+      case (lefts, focus :: rights) => some(focused(lefts, focus, rights))
+    }
+
+  def hasFocus: Boolean =
+    toDisjunction.isRight
+
+  def length: Int =
+    toDisjunction.fold(_.tail.length + 1, _.length)
+
+  def map[B](f: A => B): OptsList[B] =
+    OptsList(toDisjunction.bimap(_.map(f), _.map(f)))
+
+  def toNel: NonEmptyList[A] =
+    toDisjunction match {
+      case -\/(OneAnd(a, as)) =>
+        NonEmptyList.nel(a, as)
+      case \/-(z) =>
+        val s = z.start
+        NonEmptyList.nel(s.focus, s.rights.toList)
+    }
+
+  def toList: List[A] =
+    toDisjunction match {
+      case -\/(l) => l.toList
+      case \/-(z) => z.toList
+    }
+
+  def traverse[G[_]: Applicative, B](f: A => G[B]): G[OptsList[B]] =
+    toDisjunction match {
+      case -\/(l) => l.traverse(f).map(l => OptsList(l.left))
+      case \/-(r) => r.traverse(f).map(r => OptsList(r.right))
+    }
+
+  /** Deletes all occurrences of `a` in the options list. */
+  def delete(a: A)(implicit ev: Equal[A]): Option[OptsList[A]] =
+    toDisjunction match {
+      case -\/(l) => l.toList.filter(_ =/= a).toNel.map(nel => OptsList.unfocused(nel))
+      case \/-(z) =>
+        val l  = z.lefts.filter(_ =/= a)
+        val r  = z.rights.filter(_ =/= a)
+        val z0 = Zipper(l, z.focus, r)
+        (if (z0.focus === a) z0.delete else Some(z0)).map(zip => OptsList(zip.right))
+    }
+
+  /** Pairs each element with a boolean indicating whether that element has focus. */
+  def withFocus: OptsList[(A, Boolean)] =
+    toDisjunction match {
+      case -\/(o) => OptsList(o.strengthR(false).left)
+      case \/-(z) => OptsList(z.withFocus.right)
+    }
+}
+
+object OptsList {
+  implicit val TraverseOptsList: Traverse[OptsList] =
+    new Traverse[OptsList] {
+      def traverseImpl[G[_]: Applicative, A, B](fa: OptsList[A])(f: A => G[B]): G[OptsList[B]] =
+        fa.toDisjunction match {
+          case -\/(l) => l.traverse(f).map(l => OptsList(l.left))
+          case \/-(r) => r.traverse(f).map(r => OptsList(r.right))
+        }
+    }
```

Comment:
You usually want to override the default `map` implementation because the one defined in terms of `traverse` is kind of inefficient.